src/web/nootils.js

```diff
@@ -72,6 +73,20 @@ module.exports = function(settings) {
     isDateValid: function(date) {
       return !isNaN(date.getTime());
     },
+    getField: function(report, field) {
+      /* field is a string e.g 'dob' (equivalent to report.fields.dob) or
+      * 'screening.test_result' equivalent to report.fields.screening.test_result
+      */
+      return _.propertyOf(report.fields)(field.split('.'));
+    },
+    fieldEquals: function(report, field, value) {
+      return lib.getField(report, field) === value;
+    },
+    fieldsMatch: function(report, fieldValues) {
+      return Object.keys(fieldValues).reduce(function(agg, field) {
```

Comment:
nit: Sorry for this late thought -- but it is also possible to write this as `Object.keys(fieldValues).every(field => lib.getField(report, field) === fieldValues(field))`. Up to you what is cleaner.